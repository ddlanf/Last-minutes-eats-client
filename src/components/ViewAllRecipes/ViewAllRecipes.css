.view-all-recipes-lists{
    list-style: none;
    margin: 0;
    padding: 0;
}

.view-all-recipes-logo{
    border: 1px black solid;
    margin-top: 15vw;
    width: 20vw;
    height: 20vw;
    max-height: 200px;
    max-width: 200px;
    margin-left: auto;
    margin-right: auto;
    text-align: center;
}

.view-all-recipes-logo-food-image{
    width: 100%;
    text-align: center;
    font-size: 5vw;
}

.view-all-recipes-recipes{
    margin-top: 8%;
    margin-bottom: 8%;
    margin-right: auto;
    margin-left: auto;
    width: 90%;
}

.view-all-recipes-recipe{
    width: 100%;
    font-size: 5vw;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: stretch;
    border: 1px black solid;
}

.view-all-recipes-recipe-box{
    display: flex;
    flex-direction: column;
    width: 80%;
    max-height: 30vw;
}

.view-all-recipes-rating{
    margin: 0;
    padding-left: 2%;
    width: 100%;
    font-size: 3.5vw;
}

.view-all-recipes-recipe-name{
    margin: 0;
    text-align: center;
    font-size: 4.5vw;
    padding-left: 2%;
    padding-right: 2%;
}

.view-all-recipes-recipe-details{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}

.view-all-recipes-image-box{
    max-width: 20%;
    min-width: 20%;
    padding-right: 1%;
    padding-top: 1%;
    padding-bottom: 1%;
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    text-align: center;
}

.view-all-recipes-image{
    max-width: 100%;
    max-height: 25vw;
}

.view-all-recipes-preparation-time, .view-all-recipes-amount-of-steps, .view-all-recipes-amount-of-ingredients{
    font-size: 3.5vw;
}

a.view-all-recipes-link{
    color: black;
    text-decoration: none;
}

@media screen and (min-width: 1000px) {

    .view-all-recipes-recipes{
        width: 75%;
    }

    .view-all-recipes-recipe{
        font-size: 3vw;
    }

    .view-all-recipes-recipe-name{
        font-size: 3vw;
    }

    .view-all-recipes-preparation-time, .view-all-recipes-amount-of-steps, .view-all-recipes-amount-of-ingredients{
        font-size: 2vw;
    }

    .view-all-recipes-rating{
        font-size: 2.5vw;
    }

    .view-all-recipes-image{
        width: 100%;
        max-height: 15vw;
    }
}