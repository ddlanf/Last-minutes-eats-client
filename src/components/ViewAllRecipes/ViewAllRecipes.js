import React, { Component } from 'react'
import './ViewAllRecipes.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Link } from 'react-router-dom'
import RecipesApiService from '../../services/recipes-api-service'
import RecipeContext from '../../contexts/RecipeContext'

export default class ViewAllRecipes extends Component {

    static contextType = RecipeContext
    
    renderRecipes(){
        const { recipes } = this.context
        const allRecipes = recipes.map((recipe, index) =>{
            return (
                    <li
                        key={recipe.id}
                        >
                        <Link
                            className="view-all-recipes-link"
                            to={`/view-recipe/${recipe.id}`}>
                            <div className="view-all-recipes-recipe">
                                <div className="view-all-recipes-recipe-box"> 
                                    <p className="view-all-recipes-rating">{this.makeStars(recipe.overall_rating)}</p>
                                    <h3 className="view-all-recipes-recipe-name">{recipe.recipe_name}</h3>
                                    <div className="view-all-recipes-recipe-details">
                                        <p className="view-all-recipes-preparation-time">{recipe.preparation_time} {recipe.preparation_time_unit.slice (0, 3)}</p>
                                        <p className="view-all-recipes-amount-of-steps">{recipe.steps.length} steps</p>
                                        <p className="view-all-recipes-amount-of-ingredients">{recipe.ingredients.length} ingredients</p>
                                    </div>
                                </div>
                                <div className="view-all-recipes-image-box">
                                    <img className="view-all-recipes-image" alt={recipe.name} src={recipe.image}/>
                                </div>
                            </div>
                        </Link>
                    </li>
                    )
        })

        return allRecipes
    }

    makeStars(numberOfStars){
        const stars = []
        
        for(let i = 1; i < 6; i++){

            if(numberOfStars < i  && numberOfStars < i - 0.75){
                stars.push((<FontAwesomeIcon key={i} className="view-all-recipes-star" icon={['far', 'star']}/>))
            }
            else if(numberOfStars >= i - 0.75 && numberOfStars <= i - 0.5){
                stars.push((<FontAwesomeIcon key={i} className="view-all-recipes-star" icon="star-half-alt"/>))
            }
            else if(numberOfStars > i - 0.5 && numberOfStars < i - 0.25){
                stars.push((<FontAwesomeIcon key={i} className="view-all-recipes-star" icon="star-half-alt"/>))
            }
            else{  stars.push((<FontAwesomeIcon  key={i} className="view-all-recipes-star" icon={['fas', 'star']}/>))  }
        }

        return stars
    }

    componentDidMount(){
        RecipesApiService.getRecipes()
            .then(recipes => { this.context.setRecipes(recipes)})
            .catch(res=>{ this.context.setError(res.error)})
    }
    
    render() {
       
        return (
            <>
               <section className="view-all-recipes-recipes">
                    <ul className="view-all-recipes-lists">
                        {this.renderRecipes()}
                    </ul>
               </section>
            </>
        )
    }
}
