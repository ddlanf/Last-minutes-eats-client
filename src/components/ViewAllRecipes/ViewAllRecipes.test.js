import React from 'react';
import ReactDOM from 'react-dom';
import ViewAllRecipes from './ViewAllRecipes';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from '../../reducers/index'

describe('Make component', () => {
    
    const store = createStore(rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>   
                      <Provider store={store}>
                        <ViewAllRecipes/>
                      </Provider>
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
            <BrowserRouter>   
              <Provider store={store}>
                <ViewAllRecipes/>
              </Provider>
            </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});