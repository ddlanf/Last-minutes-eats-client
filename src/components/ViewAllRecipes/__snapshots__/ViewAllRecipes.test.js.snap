// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make component renders the UI as expected 1`] = `
<section
  className="view-all-recipes-recipes"
>
  <ul
    className="view-all-recipes-lists"
  >
    <h1
      className="view-all-recipes-loading"
    >
      Loading
      .
    </h1>
  </ul>
</section>
`;
